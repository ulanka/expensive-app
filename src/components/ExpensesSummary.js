import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesSummary from '../selectors/expenses-total'
import numeral from 'numeral';

export const ExpensesSummary = ({expenseCount,expensesTotal}) => (
    <div>
        Viewing {expenseCount} expense{expenseCount===1 ? '' : 's'} totaling {numeral(expensesTotal/100).format('$0,0.00')}
    </div>
)

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesSummary(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)

