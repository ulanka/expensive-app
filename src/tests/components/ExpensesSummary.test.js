import React from 'react';
import {shallow} from 'enzyme';
import expenses from '../fixtures/expenses';
import { ExpensesSummary } from '../../components/ExpensesSummary';


test('should render ExpensesSummary with 2 expenses totaling $94.34',() => {
    const prop = {
        expenseCount:2,
        expensesTotal:9434
    }
    const wrapper = shallow(<ExpensesSummary {...prop} />)
    expect(wrapper).toMatchSnapshot()
});

test('should render ExpensesSummary with 1 expenses totaling $94.34',() => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={9434} />)
    expect(wrapper).toMatchSnapshot()
});