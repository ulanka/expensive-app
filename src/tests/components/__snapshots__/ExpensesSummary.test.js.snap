// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expenses totaling $94.34 1`] = `
<div>
  Viewing 
  1
   expense
   totaling 
  $94.34
</div>
`;

exports[`should render ExpensesSummary with 2 expenses totaling $94.34 1`] = `
<div>
  Viewing 
  2
   expense
  s
   totaling 
  $94.34
</div>
`;
